pkg/config/k8s.go

```diff
@@ -3,16 +3,39 @@ package config
 import (
 	"context"
 	"fmt"
+	"os"
 
+	"github.com/DataDog/KubeHound/pkg/telemetry/log"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
+const (
+	clusterNameEnvVarPtr = "KH_K8S_CLUSTER_NAME_ENV_PTR"
+	clusterNameEnvVar    = "KH_K8S_CLUSTER_NAME"
+)
+
 // ClusterInfo encapsulates the target cluster information for the current run.
 type ClusterInfo struct {
 	Name string
 }
 
 func NewClusterInfo(_ context.Context) (*ClusterInfo, error) {
+	// Testing if running from pod
+	// Using an environment variable to get the cluster name as it is not provided in the pod configuration
```

Comment:
Not sure to understand the need for a pointer to another env var.
Why not directly set `KH_K8S_CLUSTER_NAME` ? Is it in case the cluster name is already defined in another env var and you want to avoid duplication? If this is the case I would argue that people generally use templating to build the deployment charts so this "deduplication" should be done over there and not here because I think here it can be quite confusing for users.

If I missed a reason, can you put somewhere in the deployment doc the meaning of those vars and how to use them?

Reply:
ok didn't see #268 
ok for documentation, but I still don't understand the need for 2 variables instead of just setting the first one

Reply:
https://github.com/DataDog/KubeHound/pull/269#discussion_r1760696985

ok now I understand the use case